lib/iris/analysis/__init__.py

```diff
@@ -1026,18 +1026,35 @@ def _percentile(data, axis, percent, **kwargs):
     If a new additive dimension is formed, then it will always be the last
     dimension of the resulting percentile data payload.
 
+    Kwargs:
+
+    * percentile_method (string) :
+        Switch to opt between the scipy.stats.mstats.mquantiles method for
+        calculating percentiles, and the much faster np.percentiles method.
+
     """
+    percentile_method = kwargs.get('percentile_method', None)
     # Ensure that the target axis is the last dimension.
     data = np.rollaxis(data, axis, start=data.ndim)
-    quantiles = np.array(percent) / 100.
     shape = data.shape[:-1]
     # Flatten any leading dimensions.
     if shape:
         data = data.reshape([np.prod(shape), data.shape[-1]])
     # Perform the percentile calculation.
-    result = scipy.stats.mstats.mquantiles(data, quantiles, axis=-1, **kwargs)
+    if percentile_method == 'fast':
+        msg = 'Cannot use fast np.percentile method with masked array.'
+        if ma.isMaskedArray(data):
+            raise TypeError(msg)
+        result = np.percentile(data, percent, axis=-1)
+    else:
+        quantiles = np.array(percent) / 100.
+        result = scipy.stats.mstats.mquantiles(data, quantiles, axis=-1,
+                                               **kwargs)
```

Comment:
As **kwargs is passed here for extended args, you should really remove any 'percentile_method' key as it could otherwise cause errors.
BTW this is also a usage mode you don't test : `cube.collapsed(..., percent=x, percentile_method=None)` : 
As stating `percentile_method=None` is not the same as not providing one, this could cause problems with other routines calling it.
The usual way I've seen elsewhere is to use `kwargs.pop` in place of `kwargs.get`.

In fact, for choice, I'd prefer distinct allowed terms here, like "numpy_percentile" and "scipy_mquantiles", and not just "None".
I guess it would be neater if it was just a method object, but as they need handling differently, I'm happy that a string modal switch makes better sense.